### PR TITLE
Configurable DSM bucket duration

### PR DESCRIPTION
**Pull request by randomanderson**
# What Does This Do

Allows the bucket duration for datastreams monitoring to be configurable

# Motivation

To support future testing. The default of 10 seconds is too slow to test things like dynamic config.

### CODE REVIEW

**Review by PerfectSlayer — COMMENTED** [review]
Hi 👋 
Do you plan to add this settings to the public documentation too?

File: internal-api/src/main/java/datadog/trace/api/Config.java [hunk 1]
```diff
@@ -2976,6 +2983,16 @@ public boolean isDataStreamsEnabled() {
...
   }
 
+  public float getDataStreamsBucketDurationSeconds() {
+    return dataStreamsBucketDurationSeconds;
+  }
+
+  public long getDataStreamsBucketDurationNanoseconds() {
+    // Rounds to the nearest millisecond before converting to nanos
+    int milliseconds = Math.round(dataStreamsBucketDurationSeconds * 1000);
+    return TimeUnit.MILLISECONDS.toNanos(milliseconds);
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply internal-api/src/main/java/datadog/trace/api/Config.java:2993]: I'm following what all the other configurable intervals do (eg `telemetryMetricsInterval`, `remoteConfigPollInterval`, etc). We want to be able to specify subsecond intervals using a float but only care about millisecond precision internally

**Review by am312 — APPROVED** [review]

File: dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java [hunk 2]
```diff
@@ -173,6 +173,7 @@ public final class ConfigDefaults {
   static final int DEFAULT_CWS_TLS_REFRESH = 5000;
 
   static final boolean DEFAULT_DATA_STREAMS_ENABLED = false;
+  public static final int DEFAULT_DATA_STREAMS_BUCKET_DURATION = 10; // seconds
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java]: Nope. Changed

**Comment by randomanderson** [comment]
> Hi 👋 Do you plan to add this settings to the public documentation too?

No, it's really just for testing. Very small bucket sizes would created additional processing load on the backend

**Review by am312 — APPROVED** [review]

**Pull request merged**